diff --git a/generic-pc/playback.cpp b/generic-pc/playback.cpp
index 3914a32..88bfea5 100644
--- a/generic-pc/playback.cpp
+++ b/generic-pc/playback.cpp
@@ -12,16 +12,45 @@ bool cPlayback::Open(playmode_t)
 void cPlayback::Close(void)
 {
 }
+       #include <stdlib.h>
+
+bool cPlayback::Start(std::string filename, std::string /*headers*/)
+{
+	static std::string tmp_filename = "";
+	static unsigned int cnt = 0;
+		char cmd[1024] = {0};
+		if(tmp_filename != filename){
+			snprintf(cmd,sizeof(cmd),"mpv_si '%s' &",filename.c_str());
+			printf("[1]###########[%s]##############\n",cmd);
+			system(cmd);
+			tmp_filename = filename ;
+			cnt = 0;
+		}else{
+			printf("REPEAT %i ########################%s#####################\n",cnt++,filename.c_str());
+			return false;
+		}
 
-bool cPlayback::Start(std::string /*filename*/, std::string /*headers*/)
-{
-	return false;
+	return true;
 }
 
 bool cPlayback::Start(char * filename, unsigned short vpid, int vtype, unsigned short apid, int ac3, unsigned int duration)
 {
+	static std::string tmp_filename = "";
+	static unsigned int cnt = 0;
 	printf("%s:%s - filename=%s vpid=%u vtype=%d apid=%u ac3=%d duration=%i\n",
 		FILENAME, __func__, filename, vpid, vtype, apid, ac3, duration);
+	char cmd[1024] = {0};
+	std::string tmp = filename;
+		if(tmp_filename != tmp){
+			snprintf(cmd,sizeof(cmd),"mpv_si '%s' &",filename);
+			printf("[2]###########[%s]##############\n",cmd);
+			system(cmd);
+			tmp_filename = filename ;
+			cnt = 0;
+		}else{
+			printf("REPEAT %i ########################%s#####################\n",cnt++,filename);
+			return false;
+		}
 	return true;
 }

