diff --git a/xbmc/cores/VideoPlayer/DVDInputStreams/DllDvdNav.h b/xbmc/cores/VideoPlayer/DVDInputStreams/DllDvdNav.h
index 52678b91a5..14376955ac 100644
--- a/xbmc/cores/VideoPlayer/DVDInputStreams/DllDvdNav.h
+++ b/xbmc/cores/VideoPlayer/DVDInputStreams/DllDvdNav.h
@@ -26,14 +26,6 @@ extern "C" {
 
  #include "dvdnav/dvdnav.h"
 
- #ifndef WIN32
- #define WIN32
- #endif // WIN32
-
- #ifndef HAVE_CONFIG_H
- #define HAVE_CONFIG_H
- #endif
-
  #include "dvdnav/dvdnav_internal.h"
  #include "dvdnav/vm.h"
  #include "dvdnav/dvd_types.h"
