--- a/common.h	2017-03-14 11:31:54.703333332 +0100
+++ b/common.h	2017-03-14 11:32:57.946666669 +0100
@@ -36,7 +36,7 @@
   do {                                                              \
      if (exp) {                                                     \
         char tmp[64];                                               \
-        esyslog("[%s,%d]: "errstr": %s", __FILE__, __LINE__,        \
+        esyslog("[%s,%d]: " errstr ": %s ", __FILE__, __LINE__,     \
                 strerror_r(errno, tmp, sizeof(tmp)));               \
         func;                                                       \
         ret;                                                        \
